composer/algorithms/seq_length_warmup/seq_length_warmup.py

```diff
@@ -309,10 +323,19 @@ def _activate_model(self, state: State, logger: Logger) -> None:
                     raise
 
             if state.auto_microbatching:
-                # Propagate across all ranks if any rank hit CUDA OOM
-                found_cuda_oom = state.device.tensor_to_device(torch.tensor([found_cuda_oom], dtype=torch.uint8))
-                dist.all_reduce(found_cuda_oom, reduce_operation='MAX')
-                if found_cuda_oom.item() == 1:
+                all_ranks_finished = False
+                while not all_ranks_finished:
+                    # Propagate across all ranks if any rank hit CUDA OOM
+                    found_cuda_oom_tensor = state.device.tensor_to_device(
+                        torch.tensor([found_cuda_oom], dtype=torch.uint8))
```

Comment:
It can be 1 if the rank OOMed -- check out the code snippet above (it's hidden by github and requires expanding) 
<img width="556" alt="image" src="https://user-images.githubusercontent.com/17102158/230171837-0f3928c8-bc67-47a5-83ce-1dd10ef67409.png">